SIGABRT after free

I cannot run `Clp` through `JuMP`, it crashes when as soon as I start the optimization. Here is a minimum example, where I am trying to minimize `c^T x` where `A x = b` and `x >=0`.
```
julia> using JuMP, Clp

julia> C =[-2.0,3.0,0.0, 0.0]
4-element Vector{Float64}:
 -2.0
  3.0
  0.0
  0.0

julia> A=[1.0 1.0 1.0 0.0;
       1.0 -1.0 0.0 1.0]
2×4 Matrix{Float64}:
 1.0   1.0  1.0  0.0
 1.0  -1.0  0.0  1.0

julia> b= [4.0,6.0]
2-element Vector{Float64}:
 4.0
 6.0

julia> model = Model(Clp.Optimizer)
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: EMPTY_OPTIMIZER
Solver name: Clp

julia> @variable(model,x[1:4]>=0)
4-element Vector{VariableRef}:
 x[1]
 x[2]
 x[3]
 x[4]

julia> @objective(model,Min,C'*x)
-2 x[1] + 3 x[2]

julia> @constraint(model,A*x.==b)
2-element Vector{ConstraintRef{Model, MathOptInterface.ConstraintIndex{MathOptInterface.ScalarAffineFunction{Float64}, MathOptInterface.EqualTo{Float64}}, ScalarShape}}:
 x[1] + x[2] + x[3] = 4.0
 x[1] - x[2] + x[4] = 6.0

julia> optimize!(model)
julia(91740,0x100cd8580) malloc: *** error for object 0xe00000000000000: pointer being freed was not allocated
julia(91740,0x100cd8580) malloc: *** set a breakpoint in malloc_error_break to debug

signal (6): Abort trap: 6
in expression starting at REPL[9]:1
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 46636441 (Pool: 46623528; Big: 12913); GC: 37
fish: Job 1, 'julia' terminated by signal SIGABRT (Abort)
```

I was trying to do a debug build for Jacob, but aarch64-apple-darwin fails with some weird linking errors at the end (the non-debug build works, only the debug one fails, why?!?) which honestly I don't have the time to look at for the next ~10 days.  In case anyone was curious:
```
/bin/sh ../../libtool --tag=CXX --mode=link c++  -g -O0 -pipe -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -Wno-unknown-pragmas -Wno-long-long   -DCLP_BUILD   -o libClp.la -rpath /workspace/destdir/lib -no-undefined -version-info 15:7:14 ClpCholeskyBase.lo ClpCholeskyDense.lo ClpConstraint.lo ClpConstraintLinear.lo ClpConstraintQuadratic.lo Clp_C_Interface.lo ClpDualRowDantzig.lo ClpDualRowPivot.lo ClpDualRowSteepest.lo ClpDummyMatrix.lo ClpDynamicExampleMatrix.lo ClpDynamicMatrix.lo ClpEventHandler.lo ClpFactorization.lo ClpGubDynamicMatrix.lo ClpGubMatrix.lo ClpHelperFunctions.lo ClpInterior.lo ClpLinearObjective.lo ClpMatrixBase.lo ClpMessage.lo ClpModel.lo ClpNetworkBasis.lo ClpNetworkMatrix.lo ClpNonLinearCost.lo ClpNode.lo ClpObjective.lo ClpPackedMatrix.lo ClpPlusMinusOneMatrix.lo ClpPredictorCorrector.lo ClpPdco.lo ClpPdcoBase.lo ClpLsqr.lo ClpPresolve.lo ClpPrimalColumnDantzig.lo ClpPrimalColumnPivot.lo ClpPrimalColumnSteepest.lo ClpQuadraticObjective.lo ClpSimplex.lo ClpSimplexDual.lo ClpSimplexNonlinear.lo ClpSimplexOther.lo ClpSimplexPrimal.lo ClpSolve.lo Idiot.lo IdiSolve.lo ClpCholeskyPardiso.lo ClpPESimplex.lo ClpPEPrimalColumnDantzig.lo ClpPEPrimalColumnSteepest.lo ClpPEDualRowDantzig.lo ClpPEDualRowSteepest.lo     ClpCholeskyMumps.lo  -L/workspace/destdir/lib -ldmumps -lzmumps -lcmumps -lsmumps -lmumps_common -lmpiseq -lpord -lmetis -lopenblas -lgfortran -lpthread -lCoinUtils  
c++ -r -keep_private_externs -nostdlib -o .libs/libClp.1.14.7.dylib-master.o  .libs/ClpCholeskyBase.o .libs/ClpCholeskyDense.o .libs/ClpConstraint.o .libs/ClpConstraintLinear.o .libs/ClpConstraintQuadratic.o .libs/Clp_C_Interface.o .libs/ClpDualRowDantzig.o .libs/ClpDualRowPivot.o .libs/ClpDualRowSteepest.o .libs/ClpDummyMatrix.o .libs/ClpDynamicExampleMatrix.o .libs/ClpDynamicMatrix.o .libs/ClpEventHandler.o .libs/ClpFactorization.o .libs/ClpGubDynamicMatrix.o .libs/ClpGubMatrix.o .libs/ClpHelperFunctions.o .libs/ClpInterior.o .libs/ClpLinearObjective.o .libs/ClpMatrixBase.o .libs/ClpMessage.o .libs/ClpModel.o .libs/ClpNetworkBasis.o .libs/ClpNetworkMatrix.o .libs/ClpNonLinearCost.o .libs/ClpNode.o .libs/ClpObjective.o .libs/ClpPackedMatrix.o .libs/ClpPlusMinusOneMatrix.o .libs/ClpPredictorCorrector.o .libs/ClpPdco.o .libs/ClpPdcoBase.o .libs/ClpLsqr.o .libs/ClpPresolve.o .libs/ClpPrimalColumnDantzig.o .libs/ClpPrimalColumnPivot.o .libs/ClpPrimalColumnSteepest.o .libs/ClpQuadraticObjective.o .libs/ClpSimplex.o .libs/ClpSimplexDual.o .libs/ClpSimplexNonlinear.o .libs/ClpSimplexOther.o .libs/ClpSimplexPrimal.o .libs/ClpSolve.o .libs/Idiot.o .libs/IdiSolve.o .libs/ClpCholeskyPardiso.o .libs/ClpPESimplex.o .libs/ClpPEPrimalColumnDantzig.o .libs/ClpPEPrimalColumnSteepest.o .libs/ClpPEDualRowDantzig.o .libs/ClpPEDualRowSteepest.o .libs/ClpCholeskyMumps.o
ldid.cpp(707): _assert(): Swap(mach_header_->filetype) == MH_EXECUTE || Swap(mach_header_->filetype) == MH_DYLIB || Swap(mach_header_->filetype) == MH_DYLINKER || Swap(mach_header_->filetype) == MH_BUNDLE
c++ -dynamiclib  -o .libs/libClp.1.14.7.dylib .libs/libClp.1.14.7.dylib-master.o  -L/workspace/destdir/lib -ldmumps -lzmumps -lcmumps -lsmumps -lmumps_common -lmpiseq -lpord -lmetis -lopenblas -lgfortran -lpthread -lCoinUtils  -install_name  /workspace/destdir/lib/libClp.1.dylib -Wl,-compatibility_version -Wl,16 -Wl,-current_version -Wl,16.7
Undefined symbols for architecture arm64:
  "__ZN17CoinIndexedVector10checkClearEv", referenced from:
      __ZN18ClpDualRowSteepest13updateWeightsEP17CoinIndexedVectorS1_S1_S1_ in libClp.1.14.7.dylib-master.o
      __ZNK16ClpFactorization20updateColumnForDebugEP17CoinIndexedVectorS1_b in libClp.1.14.7.dylib-master.o
      __ZN16ClpSimplexPrimal11pivotResultEi in libClp.1.14.7.dylib-master.o
      __ZN12ClpPESimplex22identifyCompatibleColsEiPKiP17CoinIndexedVectorS3_ in libClp.1.14.7.dylib-master.o
      __ZN12ClpPESimplex22identifyCompatibleRowsEP17CoinIndexedVectorS1_ in libClp.1.14.7.dylib-master.o
ld: symbol(s) not found for architecture arm64
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [Makefile:802: libClp.la] Error 1
make[3]: Leaving directory '/workspace/srcdir/Clp/build/Clp/src'
make[2]: *** [Makefile:714: all] Error 2
make[2]: Leaving directory '/workspace/srcdir/Clp/build/Clp/src'
make[1]: *** [Makefile:519: all-recursive] Error 1
make[1]: Leaving directory '/workspace/srcdir/Clp/build/Clp'
make: *** [Makefile:324: all-recursive] Error 1
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -15,18 +15,18 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        version: ['1.6', '1']  # Test against LTS and current minor release
+        version: ['1.10', '1']  # Test against LTS and current minor release
         os: [ubuntu-latest, macOS-latest, windows-latest]
         arch: [x64]
         include:
           # Also test against 32-bit Linux on LTS.
-          - version: '1.6'
+          - version: '1.10'
             os: ubuntu-latest
             arch: x86
           # Test against Apple M-series
-          # - version: '1'
-          #   os: macos-14
-          #   arch: aarch64
+          - version: '1'
+            os: macos-14
+            arch: aarch64
     steps:
       - uses: actions/checkout@v4
       - uses: julia-actions/setup-julia@v2

--- a/.github/workflows/format_check.yml
+++ b/.github/workflows/format_check.yml
@@ -18,7 +18,7 @@ jobs:
         shell: julia --color=yes {0}
         run: |
           using Pkg
-          Pkg.add(PackageSpec(name="JuliaFormatter", version="1"))
+          Pkg.add(PackageSpec(name="JuliaFormatter", version="2"))
           using JuliaFormatter
           format("src/MOI_wrapper", verbose=true)
           format_file("src/Clp.jl")

--- a/Project.toml
+++ b/Project.toml
@@ -1,7 +1,7 @@
 name = "Clp"
 uuid = "e2554f3b-3117-50c0-817c-e040a3ddf72d"
 repo = "https://github.com/jump-dev/Clp.jl.git"
-version = "1.2.2"
+version = "1.3.0"
 
 [deps]
 Clp_jll = "06985876-5285-5a41-9fcb-8948a742cc53"
@@ -11,12 +11,12 @@ OpenBLAS32_jll = "656ef2d0-ae68-5445-9ca0-591084a874a2"
 PrecompileTools = "aea7be01-6a6a-4083-8856-8a6e6704d82a"
 
 [compat]
-Clp_jll = "=100.1700.601, =100.1700.700, =100.1700.900, =100.1700.901"
-LinearAlgebra = "1.6"
+Clp_jll = "=100.1700.1000"
+LinearAlgebra = "1"
 MathOptInterface = "1.1"
 OpenBLAS32_jll = "0.3.10"
 PrecompileTools = "1"
-julia = "1.6"
+julia = "1.10"
 
 [extras]
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"

--- a/src/Clp.jl
+++ b/src/Clp.jl
@@ -11,23 +11,16 @@ import MathOptInterface as MOI
 import OpenBLAS32_jll
 
 function __init__()
-    @static if Sys.isapple() && Sys.ARCH == :aarch64
-        error(
-            "Clp.jl does not support Apple M-series processors. Please use HiGHS.jl instead.",
-        )
-    end
-    if VERSION >= v"1.9"
-        config = LinearAlgebra.BLAS.lbt_get_config()
-        if !any(lib -> lib.interface == :lp64, config.loaded_libs)
-            LinearAlgebra.BLAS.lbt_forward(OpenBLAS32_jll.libopenblas_path)
-        end
+    config = LinearAlgebra.BLAS.lbt_get_config()
+    if !any(lib -> lib.interface == :lp64, config.loaded_libs)
+        LinearAlgebra.BLAS.lbt_forward(OpenBLAS32_jll.libopenblas_path)
     end
     version = VersionNumber(
         Clp_VersionMajor(),
         Clp_VersionMinor(),
         Clp_VersionRelease(),
     )
-    if !(v"1.17.2" <= version <= v"1.17.9")
+    if !(v"1.17.2" <= version <= v"1.17.10")
         error(
             "You have installed version $version of Clp, which is not " *
             "supported by Clp.jl. If the version change was breaking, changes " *
@@ -50,36 +43,34 @@ end
 
 import PrecompileTools
 
-@static if !(Sys.isapple() && Sys.ARCH == :aarch64)
-    PrecompileTools.@setup_workload begin
-        PrecompileTools.@compile_workload begin
-            let
-                model = MOI.Utilities.CachingOptimizer(
-                    MOI.Utilities.UniversalFallback(
-                        MOI.Utilities.Model{Float64}(),
-                    ),
-                    MOI.instantiate(Clp.Optimizer; with_bridge_type = Float64),
-                )
-                MOI.set(model, MOI.Silent(), true)
-                x = MOI.add_variables(model, 3)
-                sets =
-                    (MOI.GreaterThan(0.0), MOI.LessThan(2.0), MOI.EqualTo(1.0))
-                for i in 1:3, f in (x[i], 1.0 * x[1] + 2.0 * x[2])
-                    MOI.supports_constraint(model, typeof(f), typeof(sets[i]))
-                    MOI.add_constraint(model, f, sets[i])
-                end
-                f = 1.0 * x[1] + x[2] + x[3]
-                MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
-                MOI.supports(model, MOI.ObjectiveFunction{typeof(f)}())
-                MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
-                MOI.optimize!(model)
-                MOI.get(model, MOI.TerminationStatus())
-                MOI.get(model, MOI.PrimalStatus())
-                MOI.get(model, MOI.DualStatus())
-                MOI.get(model, MOI.VariablePrimal(), x)
+PrecompileTools.@setup_workload begin
+    PrecompileTools.@compile_workload begin
+        let
+            model = MOI.Utilities.CachingOptimizer(
+                MOI.Utilities.UniversalFallback(
+                    MOI.Utilities.Model{Float64}(),
+                ),
+                MOI.instantiate(Clp.Optimizer; with_bridge_type = Float64),
+            )
+            MOI.set(model, MOI.Silent(), true)
+            x = MOI.add_variables(model, 3)
+            sets =
+                (MOI.GreaterThan(0.0), MOI.LessThan(2.0), MOI.EqualTo(1.0))
+            for i in 1:3, f in (x[i], 1.0 * x[1] + 2.0 * x[2])
+                MOI.supports_constraint(model, typeof(f), typeof(sets[i]))
+                MOI.add_constraint(model, f, sets[i])
             end
+            f = 1.0 * x[1] + x[2] + x[3]
+            MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+            MOI.supports(model, MOI.ObjectiveFunction{typeof(f)}())
+            MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+            MOI.optimize!(model)
+            MOI.get(model, MOI.TerminationStatus())
+            MOI.get(model, MOI.PrimalStatus())
+            MOI.get(model, MOI.DualStatus())
+            MOI.get(model, MOI.VariablePrimal(), x)
         end
     end
 end
 
-end
+end  # module

--- a/src/Clp.jl
+++ b/src/Clp.jl
@@ -47,15 +47,12 @@ PrecompileTools.@setup_workload begin
     PrecompileTools.@compile_workload begin
         let
             model = MOI.Utilities.CachingOptimizer(
-                MOI.Utilities.UniversalFallback(
-                    MOI.Utilities.Model{Float64}(),
-                ),
+                MOI.Utilities.UniversalFallback(MOI.Utilities.Model{Float64}()),
                 MOI.instantiate(Clp.Optimizer; with_bridge_type = Float64),
             )
             MOI.set(model, MOI.Silent(), true)
             x = MOI.add_variables(model, 3)
-            sets =
-                (MOI.GreaterThan(0.0), MOI.LessThan(2.0), MOI.EqualTo(1.0))
+            sets = (MOI.GreaterThan(0.0), MOI.LessThan(2.0), MOI.EqualTo(1.0))
             for i in 1:3, f in (x[i], 1.0 * x[1] + 2.0 * x[2])
                 MOI.supports_constraint(model, typeof(f), typeof(sets[i]))
                 MOI.add_constraint(model, f, sets[i])

